src/Controller/AuthController.php

```diff
@@ -93,34 +95,56 @@ public function loginPage(Request $request, Application $app)
             }
         }
 
-        $authorize_urls = $this->createAuthorizeUrls($app['auth.enabled'], $app['url_generator'], $return_url);
-        return $app['twig']->render('login.twig', $authorize_urls);
+        if (!empty($app['auth.instant_auth'])) {
+            $authorize_url = $this->createAuthorizeUrl($app['auth.instant_auth'], $app['url_generator'], $return_url);
+            return new RedirectResponse($authorize_url['url']);
+        }
+
+        return $this->loginPage($request, $app, $return_url);
+    }
+
+    private function loginPage(Request $request, Application $app, ?string $return_url)
+    {
+        $url_generator = $app['url_generator'];
+        $twig_params = array_map(function(string $auth_type) use ($url_generator, $return_url) {
+            $auth_url = $this->createAuthorizeUrl($auth_type, $url_generator, $return_url);
+            return [$auth_url['type'] . '_authorize_url' => $auth_url['url']];
+        }, $app['auth.enabled']);
+        $twig_params = array_merge(...$twig_params); // flatten
```

Comment:
flatten은 대개 reduce 과정인 경우가 많습니다.

```suggestion
        $ar = array_map(function(string $auth_type) use ($url_generator, $return_url) {
            return self::createAuthorizeUrl($auth_type, $url_generator, $return_url);
        }, $app['auth.enabled']);
        $twig_params = array_reduce($ar, function ($result, $auth_url) {
            $result[$auth_url['type'] . '_authorize_url'] = $auth_url['url'];
            return $result;
        }, []);
```

Reply:
데이터의 흐름대로 코드가 읽히게 하려면...

```php
        $twig_params = collect($app['auth.enabled'])
            ->map(function (string $auth_type) use ($url_generator, $return_url) {
                return $this->createAuthorizeUrl($auth_type, $url_generator, $return_url);
            })
            ->reduce(function ($result, $item) {
                list('type' => $type, 'url' => $url) = $item;
                $result[$type . '_authorize_url'] = $url;
                return $result;
            }, []);
```

Reply:
오 그러네요. Laravel  collections도 활용하기 좋겠어요 ㅎㅎ 감사합니다.

Reply:
적어주신 코드가 손댈데 없이 깔끔해서 그대로 커밋했습니다. 👍